### PR TITLE
Closes #7103 #5217: DownloadService: Read initial download state from store

### LINKED ISSUE #5217: Crash when downloading a large data URI
## Steps to reproduce
1. Launch sample-browser.
2. Open a website that redirects using a data URI. The one I used was: https://mars.nasa.gov/participate/send-your-name/mars2020/certificate/599173796266
3. Click download

### Expected behavior
If the Data URI is an image, a dialogue should pop up to ask the user if they want to download the image. 
### Actual behavior
Application crashes with a TransactionTooLargeException
### Device information

* Android device: **Google Pixel 3(Android 10).**

### Notes
The application redirects because ( from what I can gather), we check for the "new" url hostname versus our session, which in this case will be different. From there, we redirect with an intent. In the example above, the picture on the NASA website is ~1.2MB which will lead to the application trying to send an intent that's too big (data parcel size 3473380 bytes in my case).

### LINKED ISSUE #7103: DownloadService: Read initial download state from store
This is a first step towards https://github.com/mozilla-mobile/android-components/issues/7050.

We can add the store to the download service and read the initial download state from there, based on the download ID provided in the intent, instead of serializing the whole download state as intent extra. This will allow us to fix: https://github.com/mozilla-mobile/android-components/issues/5217

### PR DESCRIPTION
**Pull request by csadilek**
This moves the queued download state from the manager to the store, which cleans things up and also fixes #5217.

Keeping the changeset as small as possible so we can continue in #7050 and move the remaining state as well.

Tested various cases will do more now and also open PRs for R-B and Fenix.

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -13,6 +13,40 @@ permalink: /changelog/
...
+          ),
+          tabId = sessionId,
+          onNeedToRequestPermissions = { permissions ->
+              requestPermissions(permissions, REQUEST_CODE_DOWNLOAD_PERMISSIONS)
+          }),
+      owner = this,
+      view = layout
+  )
+  ```
+
```
Comment (Amejia481) [inline docs/changelog.md:49]: nit: Extra line

Comment (csadilek) [reply docs/changelog.md:45]: 👍

**Comment by NotWoods** [comment]
Since we'll be adding more properties later, should `queuedDownloads` be a map of longs to a new data class? That way we could just save `DownloadState` as a property, and move properties from `DownloadJobState` later on.

```kt
val queuedDownloads: Map<Long, ActiveDownloadState>

data class ActiveDownloadState(
  val download: DownloadState
  // add more properties in the future
)
```

**Comment by csadilek** [comment]
> Since we'll be adding more properties later, should `queuedDownloads` be a map of longs to a new data class? 

@NotWoods Yes, I thought about that too! I think it's not more work to do as a follow-up though. So I would do that when we tackle the next step. I agree that this will definitely end up being some nested structure...

**Comment by csadilek** [comment]
bors r=Amejia481

**Comment by csadilek** [comment]
bors r=Amejia481

**Pull request merged**